src/storage/txn/commands.rs

```diff
@@ -1,60 +1,212 @@
+                ),
+                _ => panic!("mismatch Op in pessimistic lock mutations"),
+            })
+            .collect();
+
+        AcquirePessimisticLock::new(
+            keys,
+            req.take_primary_lock(),
+            req.get_start_version().into(),
+            req.get_lock_ttl(),
+            req.get_is_first_lock(),
+            req.get_for_update_ts().into(),
+            WaitTimeout::from_encoded(req.get_wait_timeout()),
+            req.take_context(),
+        )
     }
+}
 
-    #[cfg(test)]
-    pub fn from_key_ts(key: Key, ts: Option<TimeStamp>) -> Self {
-        PointGetCommand {
-            ctx: Context::default(),
-            key,
-            ts,
+impl From<CommitRequest> for Command {
+    fn from(mut req: CommitRequest) -> Command {
+        let keys = req.get_keys().iter().map(|x| Key::from_raw(x)).collect();
+
+        Commit::new(
+            keys,
+            req.get_start_version().into(),
+            req.get_commit_version().into(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<CleanupRequest> for Command {
+    fn from(mut req: CleanupRequest) -> Command {
+        Cleanup::new(
+            Key::from_raw(req.get_key()),
+            req.get_start_version().into(),
+            req.get_current_ts().into(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<BatchRollbackRequest> for Command {
+    fn from(mut req: BatchRollbackRequest) -> Command {
+        let keys = req.get_keys().iter().map(|x| Key::from_raw(x)).collect();
+        Rollback::new(keys, req.get_start_version().into(), req.take_context())
+    }
+}
+
+impl From<PessimisticRollbackRequest> for Command {
+    fn from(mut req: PessimisticRollbackRequest) -> Command {
+        let keys = req.get_keys().iter().map(|x| Key::from_raw(x)).collect();
+
+        PessimisticRollback::new(
+            keys,
+            req.get_start_version().into(),
+            req.get_for_update_ts().into(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<TxnHeartBeatRequest> for Command {
+    fn from(mut req: TxnHeartBeatRequest) -> Command {
+        TxnHeartBeat::new(
+            Key::from_raw(req.get_primary_lock()),
+            req.get_start_version().into(),
+            req.get_advise_lock_ttl(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<CheckTxnStatusRequest> for Command {
+    fn from(mut req: CheckTxnStatusRequest) -> Command {
+        CheckTxnStatus::new(
+            Key::from_raw(req.get_primary_key()),
+            req.get_lock_ts().into(),
+            req.get_caller_start_ts().into(),
+            req.get_current_ts().into(),
+            req.get_rollback_if_not_exist(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<ScanLockRequest> for Command {
+    fn from(mut req: ScanLockRequest) -> Command {
+        ScanLock::new(
+            req.get_max_version().into(),
+            &req.take_start_key(),
```

Comment:
I prefer to use `Key::from_raw(req.get_start_key())` here to keep consistency with other Command. Maybe
```
let start_key = if !req.get_start_key().is_empty() {
  Some(Key::from_raw(req.get_start_key()))
} else {
  None
};
```